Student Sketches and Answers.jl

```diff
@@ -27,238 +26,320 @@ begin
-	function PLOT(animation_data, sy=zeros(maximum(animation_data.xcor)-minimum(animation_data.xcor)+1))
-		x = animation_data.xcor
-		y = animation_data.ycor
-		xmin = 0
-		xmax = 20
-		ymin = -10
-		ymax = 10
-		sx=collect(xmin:xmax)
-
-	# {{{ I reformatted a how I want you to format multi-line function calls so they are easier to read }}}
-	p=plot(sx, sy,
-			lims=(xmin, xmax), 
-			ylims=(ymin, ymax),     
-			sseriestype = :line,  # Set the seriestype to line
+# ╔═╡ 065df6ae-57e6-4eaf-9397-1ba5d37b4acc
+md"""
+## Functions and data processing
+"""
+
+# ╔═╡ 4f1000dd-0e52-40bc-8d75-b2015e3e6198
+function build_sy_dataset(a_df, s_df)
+    sy_data_1 = DataFrame(uid = Int64[], timestamp = DateTime[], y_vector = Vector{Float64}[])
+    sy_data_2 = DataFrame(uid = Int64[], timestamp = DateTime[], y_vector = Vector{Float64}[])
+
+    function update_sy_vector(student_answers, qid, sketch_data, sy_data, student, current_sketch_y_vector = zeros(22))
+        if(size(student_answers, 1) == 2 || student_answers.q_id[1] == 110)
+            end_timestamp_1 = student_answers[student_answers.q_id .== 110, :].timestamp[1]
+        else
+            end_timestamp_1 = student_answers[student_answers.q_id .== 111, :].timestamp[1]
+        end
+
+        if qid == 110
+            animation_data = filter(row -> row.timestamp <= end_timestamp_1, sketch_data)
+            current_sketch_y_vector = zeros(22)
+        else
+            if(size(student_answers, 1) == 2 || student_answers.q_id[1] == 111)
+                end_timestamp_2 = student_answers[student_answers.q_id .== 111, :].timestamp[1]
+            else
+                end_timestamp_2 = sketch_data.timestamp[length(sketch_data.timestamp)]
+            end
+            animation_data = filter(row -> (end_timestamp_1 <= row.timestamp) && (row.timestamp <= end_timestamp_2), sketch_data)
+        end
+
+        x = animation_data.xcor
+        y = animation_data.ycor
+        t = animation_data.timestamp
+
+        for i in 1:size(animation_data, 1)
+            current_sketch_y_vector[x[i] + 1] = y[i]
+            push!(sy_data.uid, student)
+            push!(sy_data.timestamp, t[i])
+            push!(sy_data.y_vector, copy(current_sketch_y_vector))
+        end
+    end
+
+    for i in 1:length(unique(a_df.uid))
+        if unique(a_df.uid)[i] == 73 || unique(a_df.uid)[i] == 87 || unique(a_df.uid)[i] == 106 || unique(a_df.uid)[i] == 100 || unique(a_df.uid)[i] == 104
+            continue
+        end
+        student_answers = a_df[a_df.uid .== unique(a_df.uid)[i], :]
+        sketch_data = filter(row -> row.uid == unique(a_df.uid)[i], s_df)
+
+        update_sy_vector(student_answers, 110, sketch_data, sy_data_1, unique(a_df.uid)[i])
+		# println(sy_data_2)
+        update_sy_vector(student_answers, 111, sketch_data, sy_data_2, unique(a_df.uid)[i],deepcopy(sy_data_1[sy_data_1.uid  .== unique(a_df.uid)[i], :].y_vector[length(sy_data_1[sy_data_1.uid  .== unique(a_df.uid)[i], :].y_vector)]))
```

Comment:
Whenever code wraps onto multiple lines, it gets hard to read. To prevent that, just put each input on its own line like you've done with plot functions.